crates/net/p2p/src/error.rs

```diff
@@ -193,9 +191,9 @@ pub enum DownloadError {
     #[display("received empty response")]
     EmptyResponse,
     /// Error while executing the request.
-    RequestError(RequestError),
+    RequestError(#[error(source)] RequestError),
     /// Provider error.
-    Provider(ProviderError),
+    Provider(#[error(source)] ProviderError),
```

Comment:
does it need the `error(source)` attribute when it's a 1-ary tuple struct?

Reply:
fixed